styles/mixins/patterns/_containers.scss

```diff
@@ -149,3 +149,65 @@ $-color-background: var.$color-background;
     }
   }
 }
+
+/**
+ * A scrolling "frame" that shows scroll-hint shadows at the top and bottom
+ * of the frame if:
+ *  - The content height exceeds the frame height: i.e. can be scrolled, and
+ *  - The content is scrollable in the shadow's direction (up or down)
+ *
+ * Shadows are not visible once the frame has been scrolled all the way in the
+ * shadow's direction. Shadows are not visible if the height of the content
+ * does not overflow the frame (is not scrollable).
+ *
+ * The shadow hinting is handled by four positioned background gradients:
+ *   - One gradient each at top and bottom of frame that obscure the shadow hints
+ *     (shadow covers). These use `background-attachment: local`, which makes
+ *     their position fixed to the _content_ within the scrollbox.
+ *   - One gradient each at the top and the bottom of the frame that are the
+ *     shadow hints (shadows). These use `background-attachment: scroll` such
+ *     that they are always positioned at the top and the bottom of the
+ *     _scrollbox_ frame. When these positions align with the positions of the
+ *     shadow covers--at the top and the bottom of the overflowing content--
+ *     they will be obscured by those shadow covers.
+ *
+ * See https://lea.verou.me/2012/04/background-attachment-local/
+ *
+ * Safari's behavior is different because of a bug with
+ * `background-attachment: local`.
+ * See https://bugs.webkit.org/show_bug.cgi?id=219324
+ * In Safari:
+ *   - Scroll-hint shadows do not appear if content does not overflow (this is
+ *     consistent with other browsers)
+ *   - Only the bottom scroll-hint shadow appears if content overflows
+ *   - The bottom scroll-hint shadow is always present, even if content is
+ *     fully scrolled
+ */
+@mixin scrollbox {
+  overflow: auto;
+  @include atoms.border;
+
+  background:
+    // Shadow covers
+    linear-gradient($-color-background 30%, rgba(255, 255, 255, 0)),
+    linear-gradient(rgba(255, 255, 255, 0), $-color-background 70%) 0 100%,
+    // Shadows
+    linear-gradient(
+        to bottom,
+        rgba(0, 0, 0, 0.1),
+        rgba(0, 0, 0, 0.05) 5px,
+        rgba(255, 255, 255, 0) 70%
+      ),
+    linear-gradient(
+        to top,
+        rgba(0, 0, 0, 0.1),
+        rgba(0, 0, 0, 0.05) 5px,
+        rgba(255, 255, 255, 0) 70%
+      )
+      0 100%;
+  background-repeat: no-repeat;
+  background-color: $-color-background;
+  background-size: 100% 40px, 100% 40px, 100% 14px, 100% 14px;
+
+  background-attachment: local, local, scroll, scroll;
+}
```

Comment:
For the effect to work, if I understand correctly, the shadow covers background layer need to be rendered above the shadows. I was surprised when reading this because the shadow covers layer was declared first, and in the DOM later elements are stacked above earlier ones by default. Apparently [CSS backgrounds](https://css-tricks.com/stacking-order-of-multiple-backgrounds/) don't work that way though.

Reply:
Yes, me too! It seemed backwards.